### PR TITLE
Port `ArraySort` to `function-arrays` subcrate

### LINKED ISSUE #9550: Port `ArraySort` to `function-arrays` subcrate
### Is your feature request related to a problem or challenge?

ArraySort function needs to be ported to new `function-arrays` subcreate in terms of #9285.

### Describe the solution you'd like

Port `ArraySort` to `function-arrays` subcrate

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by erenavsarogullari**
## Which issue does this PR close?
Closes #9550

## What changes are included in this PR?
This PR aims to do following changes in terms of Epic https://github.com/apache/arrow-datafusion/issues/9285:
1- `ArraySort` is ported to `function-arrays` subcrate,
2- `ScalarUDFImpl. return_type` has been implemented to support input types: `List`, `FixedSizeList` and `LargeList`,
3- Function Documentation is updated.

## Are these changes tested?
Yes, all `array.slt` based `array_sort` tests are passed.

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/sqllogictest/test_files/aggregate.slt [hunk 1]
```diff
@@ -182,7 +182,7 @@ CREATE TABLE array_agg_distinct_list_table AS VALUES
   ('b', [0,1])
 ;
 
-# Apply array_sort to have determinisitic result, higher dimension nested array also works but not for array sort,
+# Apply array_sort to have deterministic result, higher dimension nested array also works but not for array sort,
```
Comment (jayzhan211) [inline datafusion/sqllogictest/test_files/aggregate.slt:185]: 👍

File: datafusion/functions-array/src/udf.rs [hunk 2]
```diff
@@ -286,6 +286,70 @@ impl ScalarUDFImpl for ArrayDims {
...
+impl crate::udf::ArraySort {
+    pub fn new() -> Self {
+        Self {
+            signature: Signature::variadic_any(Volatility::Immutable),
+            aliases: vec!["array_sort".to_string(), "list_sort".to_string()],
+        }
+    }
+}
+
+impl ScalarUDFImpl for crate::udf::ArraySort {
```
Comment (jayzhan211) [inline datafusion/functions-array/src/udf.rs:312]: +1

**Review by jayzhan211 — APPROVED** [review]
LGTM

File: datafusion/functions-array/src/udf.rs [hunk 3]
```diff
@@ -286,6 +286,70 @@ impl ScalarUDFImpl for ArrayDims {
...
+    array_sort_udf
+);
+
+#[derive(Debug)]
+pub(super) struct ArraySort {
+    signature: Signature,
+    aliases: Vec<String>,
+}
+
+impl crate::udf::ArraySort {
```
[the remark this answers is not in the document]
Comment (erenavsarogullari) [reply datafusion/functions-array/src/udf.rs]: @jayzhan211 Thanks for the review. Addressed.

File: datafusion/functions-array/src/udf.rs (hunk 2 above)
Comment (erenavsarogullari) [reply datafusion/functions-array/src/udf.rs]: Addressed

**Comment by jayzhan211** [comment]
Thanks @erenavsarogullari

**Pull request merged**